File: core/src/main/java/com/mcreater/amclcore/java/JavaEnvironment.java
package com.mcreater.amclcore.java;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.io.File;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class JavaEnvironment {
    @Getter
    private File executable;

    public static JavaEnvironment create(File executable) {
        return new JavaEnvironment(executable);
    }
}

File: core/src/main/java/com/mcreater/amclcore/java/JavaEnvironment.kt
package com.mcreater.amclcore.java

import java.io.File


class JavaEnvironment private constructor(private val executable: File? = null) {
    fun getExecutable(): File? {
        return executable
    }

    companion object {
        @JvmStatic
        fun create(executable: File?): JavaEnvironment {
            return JavaEnvironment(executable)
        }
    }
}

